cmd/hc-install/cmd_install.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/mitchellh/cli"
+	"gophers.dev/pkgs/ignore"
+
+	hci "github.com/hashicorp/hc-install"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/hc-install/src"
+)
+
+type InstallCommand struct {
+	Ui cli.Ui
+}
+
+func (c *InstallCommand) Name() string { return "install" }
+
+func (c *InstallCommand) Synopsis() string {
+	return "Install a HashiCorp product"
+}
+
+func (c *InstallCommand) Help() string {
+	helpText := `
+Usage: hc-install install [options] -version <version> <product>
+
+  This command installs a HashiCorp product.
+  Options:
+    -version  [REQUIRED] Version of product to install.
```

Comment:
What if you want to install the latest version (either latest stable (default) or latest prerelease)?

Reply:
I don't have any too strong opinion on this.

Personally I'd be fine with keeping this required initially and flipping it optional later, which is how one could express _latest stable_? I'm not sure if there's a strong need for _latest pre-release_ in the CLI. If there is, we could probably solve that with another flag - say `-prerelease` which would conflict with `-version` and map directly to https://github.com/hashicorp/hc-install/blob/5bf52541bb816aa9364a71733940c9842e01ba54/releases/latest_version.go#L26

So I think this is solvable later without breaking the API, but if you feel strongly about supporting it now - I'm okay with that as well! 😃